### PR TITLE
cleanup: update test beforeAll

### PR DESCRIPTION
**Pull request by xtineskim**
## Background
In this PR https://github.com/googleapis/java-container/issues/656 we saw that the quota for our clusters hit a max since tests were not cleaning up after themselves. Per Neenu's suggestion, we added a cleanup to the `beforeAll` to ensure that old clusters are deleted

### CODE REVIEW

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class Util {
+  // Cleans existing test resources if any.
+  private static final int DELETION_THRESHOLD_TIME_HOURS = 24;
+
+  /** tear down any clusters that are older than 24 hours * */
```
Comment (Shabirmean) [inline google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:33]: Nit: `* */` -> `**/`

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    List<Cluster> clusters = clustersResponse.getClustersList();
+
+    for (Cluster cluster : clusters) {
+      if (isCreatedBeforeThresholdTime(cluster.getCreateTime())) {
+        client.deleteCluster(projectId, zone, cluster.getName());
+      }
+    }
+  }
+
+  public static boolean isCreatedBeforeThresholdTime(String timestamp) {
```
Comment (Shabirmean) [inline google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:48]: Can we make this method private?

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  public static void cleanUpExistingInstanceCluster(
+      String projectId, String zone, ClusterManagerClient client)
+      throws IOException, ExecutionException, InterruptedException {
+
+    ListClustersResponse clustersResponse = client.listClusters(projectId, zone);
+    List<Cluster> clusters = clustersResponse.getClustersList();
+
+    for (Cluster cluster : clusters) {
+      if (isCreatedBeforeThresholdTime(cluster.getCreateTime())) {
+        client.deleteCluster(projectId, zone, cluster.getName());
```
Comment (Shabirmean) [inline google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:43]: Is [this](https://cloud.google.com/java/docs/reference/google-cloud-container/latest/com.google.cloud.container.v1beta1.ClusterManagerClient#com_google_cloud_container_v1beta1_ClusterManagerClient_deleteCluster_java_lang_String_java_lang_String_java_lang_String_) an async operation? should we later check if the cluster properly got deleted before proceeding?

**Review by Shabirmean — CHANGES_REQUESTED** [review]
Some minor comments and suggestions. Is there a way we can test? Are we creating clusters in the JDST project and re-triggering the CI on this PR?

**Review by Neenu1995 — APPROVED** [review]

**Comment by Neenu1995** [comment]
LGTM. But address Shabir's comments before merging.

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java (hunk 1 above)
Comment (ckim328) [reply google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:33]: eyes of an eagle!

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java (hunk 2 above)
Comment (Shabirmean) [reply google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:48]: I see it still being `public` in the latest commit. Do we want to stick to it. If it's not  used outside, Id change it to private.

**Review by Shabirmean — APPROVED** [review]
LGTM!

I see that the latest commits haven't still updated the code based on the comments. Maybe leave a feedback if you choose to not change them

File: google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java (hunk 3 above)
Comment (Shabirmean) [reply google-cloud-container/src/test/java/com/google/cloud/container/v1/it/Util.java:43]: As discussed offline, this is not necessary to be handled in this PR

**Pull request merged**